tests/integration/components/sl-calendar-test.js

```diff
@@ -0,0 +1,1461 @@
+});
+
+test( 'Navigating Backward by Month Crosses to Previous Year', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 1;
+    this.set( 'currentMonth', currentMonth );
+
+    this.render( hbs`
+        {{sl-calendar
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '>:first-child .datepicker-switch' ).text().trim(),
+        'January ' + currentYear,
+        'The current month is set correctly'
+    );
+
+    this.$( '>:first-child .prev' ).click();
+
+    assert.equal(
+        this.$( '>:first-child .datepicker-switch' ).text().trim(),
+        'December ' + ( currentYear - 1 ),
+        'The previous month is in the previous year'
+    );
+});
+
+test( 'All Twelve Months are Displayed in Order', function( assert ) {
+
+    this.render( hbs`
+        {{sl-calendar
+            viewMode="months"
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '>:first-child .month' ).text().trim(),
+        'JanFebMarAprMayJunJulAugSepOctNovDec',
+        'Twelve months are listed in order'
+    );
+});
+
+test( 'Twelve Years are Displayed in Order', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    this.render( hbs`
+        {{sl-calendar
+            viewMode="years"
+            currentYear=currentYear
+        }}
+    ` );
+
+    let yearsSelection = this.$( '>:first-child' ).find( 'td' ).text().trim();
+    yearsSelection = yearsSelection.replace( /\s+/g, '' );
+
+    assert.equal(
+        yearsSelection,
+        '201920202021202220232024202520262027202820292030',
+        'Twelve years are listed in order'
+    );
+});
+
+// ******************************
+// Start of Dual Component Tests
+// ******************************
+
+test( 'Dual instance: Action fires when day is clicked', function( assert ) {
+
+    const currentYear1 = 2035;
+    this.set( 'currentYear1', currentYear1 );
+
+    const currentMonth1 = 1;
+    this.set( 'currentMonth1', currentMonth1 );
+
+    const currentYear2 = 2022;
+    this.set( 'currentYear2', currentYear2 );
+
+    const currentMonth2 = 9;
+    this.set( 'currentMonth2', currentMonth2 );
+
+    this.set( 'content', testContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action="testAction1"
+            content=content
+            currentYear=currentYear1
+            currentMonth=currentMonth1
+        }}
+        {{sl-calendar
+            action="testAction2"
+            content=content
+            currentYear=currentYear2
+            currentMonth=currentMonth2
+        }}
+    ` );
+
+    this.on( 'testAction1', () => {
+        assert.ok(
+            true,
+            'The test action 1 was called'
+        );
+    });
+
+    this.on( 'testAction2', () => {
+        assert.ok(
+            true,
+            'The test action 2 was called'
+        );
+    });
+
```

Comment:
Convert these `.on()` calls and assertions to spies so that it can be determined whether "testAction1" was called.